http/src/client/mod.rs

```diff
@@ -2472,75 +2471,38 @@ impl Client {
 
             if let Some(default_headers) = &self.default_headers {
                 for (name, value) in default_headers {
-                    headers.insert(name, HeaderValue::from(value));
+                    headers.insert(name, value.clone());
                 }
             }
         }
 
         let req = if let Some(form) = form {
-            let form_bytes = form.build();
-
-            if let Some(headers) = builder.headers_mut() {
-                headers.insert(CONTENT_LENGTH, HeaderValue::from(form_bytes.len()));
-            };
-
-            builder
-                .body(Body::from(form_bytes))
-                .map_err(|source| Error {
-                    kind: ErrorType::BuildingRequest,
-                    source: Some(Box::new(source)),
-                })?
+            builder.body(Body::from(form.build()))
         } else if let Some(bytes) = body {
-            builder.body(Body::from(bytes)).map_err(|source| Error {
-                kind: ErrorType::BuildingRequest,
-                source: Some(Box::new(source)),
-            })?
-        } else if method == Method::Put || method == Method::Post || method == Method::Patch {
-            if let Some(headers) = builder.headers_mut() {
-                headers.insert(CONTENT_LENGTH, HeaderValue::from(0));
-            }
-
-            builder.body(Body::empty()).map_err(|source| Error {
-                kind: ErrorType::BuildingRequest,
-                source: Some(Box::new(source)),
-            })?
+            builder.body(Body::from(bytes))
         } else {
-            builder.body(Body::empty()).map_err(|source| Error {
-                kind: ErrorType::BuildingRequest,
-                source: Some(Box::new(source)),
-            })?
-        };
+            builder.body(Body::empty())
+        }
+        .map_err(|source| Error {
+            kind: ErrorType::BuildingRequest,
+            source: Some(Box::new(source)),
+        })?;
```

Comment:
It's hard to read the flow of this if-else chain. I feel we have a two options we could go with here:

1. name `req` as `try_req` and then do `let req = try_req.map_err(...)?;`
2. split this out to a function, eg `let req = set_body(...).map_err(...)?;`

Reply:
I think the solution I went with is quite good, let me know what you think

Reply:
lgtm